R/arx_forecaster.R

```diff
@@ -83,28 +87,35 @@ arx_forecaster <- function(x, y, key_vars, time_value,
 #' arx_args_list()
 #' arx_args_list(symmetrize = FALSE)
 #' arx_args_list(levels = c(.1, .3, .7, .9), min_train_window = 120)
-arx_args_list <- function(lags = c(0, 7, 14), ahead = 7, min_train_window = 20,
-                          levels = c(0.05, 0.95), intercept = TRUE,
+arx_args_list <- function(lags = c(0L, 7L, 14L),
```

Comment:
Is there a reason why `intercept` was removed? Is it because it seems like we would always have the intercept in the ARX model no matter wha?

Reply:
Should we necessarily have these default `lags` and `ahead` arguments? The defaults seem specifically defined for `time_type` = `day` but there are many other time types for which the defaults make a bit less sense

Reply:
If we do leave the defaults as is, it may be good to briefly mention the defaults somewhere else in `arx_args_list()` aside from the initial list of arguments to emphasize them (under the argument descriptions seems to be natural choice, where I see mention of defaults for some but not all of the arguments).

Reply:
On Intercept, yes, it's always there.

On the `time_type`. Good point. Our implementation always modifies the `time_value` by a day. 

Actually, can someone create an issue to:
1. adjust the lead/lag function to accept a `time_unit` argument (days, months, weeks, years) and perform the operations appropriately.
2. Add an arg here (and elsewhere) to allow a user to specify the units (possibly different for lags/ahead/train_window)